src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -95,6 +95,101 @@ public void Kill()
             }
         }
 
+        /// <summary>
+        /// Makes a best-effort attempt to kill all descendant (child, grandchild, etc.) processes.
+        /// </summary>
+        private void KillTree()
+        {
+            try
+            {
+                // Causes this object instance to hold a handle to the process. While held, the process's PID won't be reused 
+                // and its children can be enumerated. These behaviors hold true even if the process is subsequently terminated.
+                OpenProcessHandle();
+
+                // Kill the process, so that no further children can be created.
+                Kill();
+            }
+            catch
+            {
+                // Made a best-effort attempt which failed (perhaps because the process is already dead), so give up.
+                return;
```

Comment:
Should we still be looking for children if the parent fails?

Reply:
Certainly of course child processes often live longer than their parents.

Reply:
Now, if `Kill()` fails, we still to terminate child processes.